### PR TITLE
Add unit tests for P2P Outbound peer class - Closes #4225

### LINKED ISSUE #4225: Add unit test coverage for P2P outbound peer file
### Expected behavior

Every file from `lisk-p2p` library is well covered with unit test. The minimum must be 80% at least to achieve green reports.  

### Actual behavior

`peer/outbound.ts` counts with only  52.38 %.

### Steps to reproduce

`npm t`

### Which version(s) does this affect? (Environment, OS, etc...)

`+2.3`

### PR DESCRIPTION
**Pull request by diego-G**
### How to manually test it?

`npm t`

### Review checklist

- [x] The PR resolves #4225
- [x] All new code is covered with unit tests
- [x] Relevant integration / functional tests are added
- [x] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [x] Documentation has been added/updated

### CODE REVIEW

File: elements/lisk-p2p/test/unit/peer/outbound.ts [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+	beforeEach(() => {
+		defaultOutboundPeer = new OutboundPeer(
+			defaultPeerInfo,
+			defaultOutboundPeerConfig,
+		);
+	});
+
+	afterEach(() => {
+		sandbox.restore();
```
Comment (shuse2) [inline elements/lisk-p2p/test/unit/peer/outbound.ts]: restore should be called by default in `_global_hook.ts`

File: elements/lisk-p2p/test/unit/peer/outbound.ts [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+				defaultOutboundPeer as any,
+				'_unbindHandlersFromOutboundSocket',
+			);
+
+			(defaultOutboundPeer as any)._socket = outboundSocket;
+			expect((defaultOutboundPeer as any)._socket).to.eql(outboundSocket);
+			defaultOutboundPeer.socket = outboundSocket;
+			expect((defaultOutboundPeer as any)._unbindHandlersFromOutboundSocket).to
+				.be.calledOnce;
+		});
```
Comment (shuse2) [inline elements/lisk-p2p/test/unit/peer/outbound.ts]: prefer new line in between `it`

File: elements/lisk-p2p/test/unit/peer/outbound.ts [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+		it('should set new socket', () => {
+			const outboundSocket = socketClusterClient.create(clientOptions);
```
Comment (shuse2) [inline elements/lisk-p2p/test/unit/peer/outbound.ts]: doesn't this socket hangs around if we create the real socket? same for other cases.
Maybe it's good to mock it.

File: elements/lisk-p2p/test/unit/peer/outbound.ts [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+				.returns(outboundSocket);
+
+			expect((defaultOutboundPeer as any)._socket).to.be.undefined;
+			defaultOutboundPeer.send(packet);
+			expect((defaultOutboundPeer as any)._createOutboundSocket).to.be
+				.calledOnce;
+			expect((defaultOutboundPeer as any)._socket).to.eql(outboundSocket);
+		});
+
+		it('should call send and emit event', () => {
```
Comment (shuse2) [inline elements/lisk-p2p/test/unit/peer/outbound.ts]: should call `emit`?

File: elements/lisk-p2p/test/unit/peer/outbound.ts [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+
+			(defaultOutboundPeer as any)._socket = outboundSocket;
+			defaultOutboundPeer.socket = outboundSocket;
+			expect((defaultOutboundPeer as any)._bindHandlersToOutboundSocket).to.be
+				.calledOnce;
+		});
+	});
+
+	describe('#connect', () => {
+		it('should create outbound socket if it does not exist', () => {
```
Comment (shuse2) [inline elements/lisk-p2p/test/unit/peer/outbound.ts]: also good to check opposite case too? shouldnt create if it already exist

File: elements/lisk-p2p/test/unit/peer/outbound.ts [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+		defaultOutboundPeer.disconnect();
+	});
+
+	describe('#constructor', () => {
+		it('should be an instance of OutboundPeer class', () =>
+			expect(defaultOutboundPeer).to.be.instanceof(OutboundPeer));
+	});
+
+	describe('#socket', () => {
+		it('should unbind handlers from outbound socket if it exists', () => {
```
Comment (shuse2) [inline elements/lisk-p2p/test/unit/peer/outbound.ts]: same as below, probably it's good to check the opposite case

File: elements/lisk-p2p/test/unit/peer/outbound.ts [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+
+			(defaultOutboundPeer as any)._socket = outboundSocket;
+			expect((defaultOutboundPeer as any)._socket).to.eql(outboundSocket);
+			defaultOutboundPeer.socket = outboundSocket;
+			expect((defaultOutboundPeer as any)._unbindHandlersFromOutboundSocket).to
+				.be.calledOnce;
+		});
+
+		it('should set new socket', () => {
+			expect((defaultOutboundPeer as any)._socket).to.be.undefined;
```
[the remark this answers is not in the document]
Comment (diego-G) [reply elements/lisk-p2p/test/unit/peer/outbound.ts]: You're right but I did it to avoid the possibility of having false positives. It doesn't bother us.

**Pull request merged**